#

sandeepl337 opened the pull request:
###

Commit:
[VarDumper] fixed HtmlDumper to target specific the head tag

fabpot committed:
moved PHP nightly to PHP 7.0

leofeyer committed:
Support omitting the <target> node in an .xlf file.

fabpot committed:
minor #15598 moved PHP nightly to PHP 7.0 (fabpot)

This PR was merged into the 2.3 branch.

Discussion
----------

moved PHP nightly to PHP 7.0

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Commits
-------

40e3ab0 moved PHP nightly to PHP 7.0

stof committed:
Fix the handling of values for multiple choice types

Choice values must always be strings, but a place was missing the
casting, breaking the comparison of selected choices when the callback
does not return a string.

fabpot committed:
bug #15608 [Form] Fix the handling of values for multiple choice types (stof)

This PR was merged into the 2.7 branch.

Discussion
----------

[Form] Fix the handling of values for multiple choice types

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Choice values must always be strings (the ChoiceListInterface documents this requirement), but a place was missing the casting, breaking the comparison of selected choices when the callback does not return a string. This happens when a property path is used and points to a non-string property for instance (in the tests, the id is an integer, which is common).
Single choice types are not affected by the bug because the data transformer was also taking care of the string casting in this case.

Commits
-------

f31fa8c Fix the handling of values for multiple choice types

aitboudad committed:
bug #15611 [Translation][Xliff Loader] Support omitting the <target> node in an .xlf file. (leofeyer)

This PR was merged into the 2.3 branch.

Discussion
----------

[Translation][Xliff Loader] Support omitting the <target> node in an .xlf file.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

This is a cherry pick of #15604 on 2.3

Commits
-------

3dcda1a Support omitting the <target> node in an .xlf file.

aitboudad committed:
Merge branch '2.3' into 2.7

Conflicts:
	.travis.yml
	src/Symfony/Component/Translation/Loader/XliffFileLoader.php

WouterJ commented:
@sandeepl337 please remove the description here and sent a mail to security [at] symfony.com as described [here](http://symfony.com/doc/current/contributing/code/security.html). You should never publicly report security issues

sandeepl337 commented:
Sorry for that! Done! I'm sending to Security Email.

nicolas-grekas committed:
bug #15493 [VarDumper] fixed HtmlDumper to target specific the head tag (SaschaDens)

This PR was submitted for the 2.6 branch but it was merged into the 2.7 branch instead (closes #15493).

Discussion
----------

[VarDumper] fixed HtmlDumper to target specific the head tag

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

On our webportals we faced the issue that we received the following js error: `Uncaught HierarchyRequestError: Failed to execute 'appendChild' on 'Node': This node type does not support this method.` and then the full dump is displaying without the arrows to collapse.

To simulate this error I've created a [Gist](https://gist.github.com/SaschaDens/0c5cc610ba168f6206d2) and as reference used [Adventures With document.documentElement.firstChild](https://robert.accettura.com/blog/2009/12/12/adventures-with-document-documentelement-firstchild/)

This PR specifically targets the head by using `document.getElementsByTagName('head')[0]` instead of `document.documentElement.firstChild`

Commits
-------

cbba074 [VarDumper] fixed HtmlDumper to target specific the head tag

WouterJ commented:
Thanks @sandeepl337! Can you (or any maintainer) please press the close button for this PR?
